src/ExplainAnalyzer.php

```
@@ -126,16 +126,25 @@ public function analyze(array $explainResult, array $warnings = []): array
         $detectedWarnings = [];
         foreach ($this->warnings as $warningType => $warning) {
             if (isset($warning['detector'])) {
-                if ($warning['detector']->detect($explainResult)) {
-                    $detectedWarnings[] = ['type' => $warningType, 'message' => $warning['message'], 'documentation' => $this->getDocumentationUrl($warningType)];
+                $detector = $warning['detector'];
+                if ($detector->detect($explainResult)) {
```

PHPStan (2.2.1)

[error] Parameter `#1` $explainResult of method Koriym\SqlQuality\Detector\DetectorInterface::detect() expects array{query_block: array{select_id: int, cost_info?: array{query_cost?: float, read_cost?: float, eval_cost?: float, sort_cost?: float, tmp_table_cost?: float}, table?: array{table_name: string, access_type: string, possible_keys?: string|null, key?: string|null, rows: int, rows_examined_per_scan?: int, filtered?: float, attached_condition?: string, ...}, grouping_operation?: array{using_temporary_table?: bool, using_filesort?: bool, cost_info?: array{query_cost?: float, read_cost?: float, eval_cost?: float, sort_cost?: float, tmp_table_cost?: float}, table?: array{table_name: string, access_type: string, possible_keys?: string|null, key?: string|null, rows: int, rows_examined_per_scan?: int, filtered?: float, attached_condition?: string, ...}, nested_loop?: array<array{table: array{table_name: string, access_type: string, possible_keys?: string|null, key?: string|null, rows: int, rows_examined_per_scan?: int, filtered?: float, attached_condition?: string, ...}}>, duplicates_removal?: array{using_temporary_table?: bool, using_filesort?: bool, table?: array{table_name: string, access_type: string, possible_keys?: string|null, key?: string|null, rows: int, rows_examined_per_scan?: int, filtered?: float, attached_condition?: string, ...}}}, ordering_operation?: array{using_temporary_table?: bool, using_filesort?: bool, cost_info?: array{query_cost?: float, read_cost?: float, eval_cost?: float, sort_cost?: float, tmp_table_cost?: float}, table?: array{table_name: string, access_type: string, possible_keys?: string|null, key?: string|null, rows: int, rows_examined_per_scan?: int, filtered?: float, attached_condition?: string, ...}, nested_loop?: array<array{table: array{table_name: string, access_type: string, possible_keys?: string|null, key?: string|null, rows: int, rows_examined_per_scan?: int, filtered?: float, attached_condition?: string, ...}}>, duplicates_removal?: array{using_temporary_table?: bool, using_filesort?: bool, table?: array{table_name: string, access_type: string, possible_keys?: string|null, key?: string|null, rows: int, rows_examined_per_scan?: int, filtered?: float, attached_condition?: string, ...}}}, duplicates_removal?: array{using_temporary_table?: bool, using_filesort?: bool, table?: array{table_name: string, access_type: string, possible_keys?: string|null, key?: string|null, rows: int, rows_examined_per_scan?: int, filtered?: float, attached_condition?: string, ...}}, select_list_subqueries?: array<array{query_block: array{table: array{table_name: string, access_type: string, possible_keys?: string|null, key?: string|null, rows: int, rows_examined_per_scan?: int, filtered?: float, attached_condition?: string, ...}}}>, nested_loop?: array{using_temporary_table?: bool, using_filesort?: bool, cost_info?: array{query_cost?: float, read_cost?: float, eval_cost?: float, sort_cost?: float, tmp_table_cost?: float}, table?: array{table_name: string, access_type: string, possible_keys?: string|null, key?: string|null, rows: int, rows_examined_per_scan?: int, filtered?: float, attached_condition?: string, ...}, nested_loop?: array<array{table: array{table_name: string, access_type: string, possible_keys?: string|null, key?: string|null, rows: int, rows_examined_per_scan?: int, filtered?: float, attached_condition?: string, ...}}>, duplicates_removal?: array{using_temporary_table?: bool, using_filesort?: bool, table?: array{table_name: string, access_type: string, possible_keys?: string|null, key?: string|null, rows: int, rows_examined_per_scan?: int, filtered?: float, attached_condition?: string, ...}}}, ...}, analyze_result: array<string, mixed>}, array given.


(argument.type)